src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -1189,9 +1208,9 @@ public SchemaShape GetOutputSchema(SchemaShape inputSchema)
                 var input = _options.InputColumns[i];
                 if (!inputSchema.TryFindColumn(input, out var col))
                     throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input);
-                var expectedType = DnnUtils.Tf2MlNetType(_tfInputTypes[i]);
-                if (col.ItemType != expectedType)
-                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input, expectedType.ToString(), col.ItemType.ToString());
+                //var expectedType = DnnUtils.Tf2MlNetType(_tfInputTypes[i]);
+                //if (col.ItemType != expectedType)
+                //    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input, expectedType.ToString(), col.ItemType.ToString());
```

Comment:
Why is this being commented out?

Reply:
These are checks with input types. I made changes to check for Vector type in my most recent commit.

Reply:
`col.ItemType` is a `PrimitiveDataViewType`, which is the type of the elements in the vector, so you should not compare it to the vector type, but to the item type.
Additionally, it seems like there is a discrepancy between how `_inputTypes` is defined (it is always an array of length 1), and the fact that the input columns can be passed as an array in the `Options` object. The way I understand it, is there should be exactly two input columns to the estimator: the label column, and the column containing the image bytes. These should be passed as two strings in the `Options` object (as opposed to an array of strings), and these are the columns whose types should be checked in this method.

(I have also commented about this in PR #4228, just FYI, so you don't make conflicting changes...)

---
In reply to: [328356487](https://github.com/dotnet/machinelearning/pull/4242#discussion_r328356487) [](ancestors = 328356487)